cmd/ipfs/pinmfs.go

```diff
@@ -227,6 +225,10 @@ func pinMFS(
 			pinTime = ps.GetCreated().UTC()
 			break
 		}
+		//Run a goroutine to monitor status of queued pins
+		if ps.GetPin().GetCid() == cid && ps.GetStatus() == pinclient.StatusQueued {
```

Comment:
The code above already does `if ps.GetPin().GetCid() == cid` so this and the above should be under a that single `if` condition.